src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/NodeUtil.java

```diff
@@ -15,12 +15,23 @@ private NodeUtil() {}
      * @return <code>true</code> if the job is ok to be used
      */
     public static boolean isNodeOnline(String nodeName) {
-        String controllerLabel = Jenkins.get().getSelfLabel().getName();
+        return isNodeOnline(nodeName, Jenkins.get());
+    }
+
+    /**
+     * Checks whether the given node is available for an execution of the job,
+     *
+     * @param nodeName the name of the node to check
+     * @param jenkins  the Jenkins instance to use
+     * @return <code>true</code> if the job is ok to be used
+     */
+    public static boolean isNodeOnline(String nodeName, Jenkins jenkins) {
+        String controllerLabel = jenkins.getSelfLabel().getName();
         if (controllerLabel.equals(nodeName)) {
             return true;
         }
 
-        final Computer c = Jenkins.get().getComputer(nodeName);
+        final Computer c = jenkins.getComputer(nodeName);
```

Comment:
Not needed to change the method in order to test it.
```suggestion
        final Computer c = Jenkins.get().getComputer(nodeName);
```